### ISSUE TITLE
Results are sometimes dropped when batched and unbatched resolvers occur in the same query

### ISSUE THREAD

**Issue by eneroth** [wilkerlucio/pathom3#37]
This is going to be hard to produce a minimally working example of, but perhaps we can show you live.

There are two resolvers, `sub-atoms` and `particles`. `sub-atoms` is batched, `particles` is not. When a query is run which involves both, it returns:

```clojure 
    :particles [{} {}],
    :sub-atoms []}
```

(`sub-atoms` should be empty here, but the particles should not be empty)

When we make `particles` a batch resolver, they resolve as intended.

**Comment by avocade** [comment]
Also, when `sub-atoms` is _not_ an empty vec (when it has values), the particles resolver works as expected and has the correct values as well. Very weird, feels like some sort of planner/execution misbehaviour.

**Comment by wilkerlucio** [comment]
Hello guys,

I'm suspecting you having the same issue described here: https://github.com/wilkerlucio/pathom3/issues/30

TLDR: batch is broken when it's part of a "group" that needs further processing.

This is high on my priority to fix, I'm finishing a blog post to talk about some planner fixes, and after that, I should jump on this batching issue.

If you believe your problem is different, please give me an example and I'll add this batch fix process.

**Comment by eneroth** [comment]
Cheers @wilkerlucio, it sure sounds like the same problem.
